dagshub/auth/tokens.py

```diff
@@ -360,3 +371,24 @@ def add_oauth_token(host: Optional[str] = None, **kwargs):
     host = host or config.host
     token = oauth.oauth_flow(host)
     _get_token_storage(**kwargs).add_token(token, host, skip_validation=True)
+
+
+def add_permanent_token(host: Optional[str] = None, **kwargs):
+    token = get_token_object(host=host, **kwargs)
+    # todo: if a permanent token already exists and is valid, skip the next step
+    perm = _generate_permanent_token(token)
```

Comment:
Check that `token` is of type `OAuthDagshubToken`, that should be enough
Alternatively add an `is_expirable()` onto the Token ABC, but that's probably overkill